### ISSUE TITLE
Krita: Upstream patches

### ISSUE THREAD

**Issue by n0toose** [haikuports/haikuports#7918]
The current version of Krita (as of the time of this writing, [4.4.8](https://github.com/haikuports/haikuports/tree/4b41fc7a40c88d9266f7b52fcfad44b7d31938fe/media-gfx/krita)) is, according to the maintainer, in response to a user trying it out on a Thinkpad X200-based Haiku installation, "quite ancient".

@hallarempt expressed on Mastodon that she'd be willing to accept any Haiku patches, as well as test Krita on Haiku every now and then. I had a very pleasant IRC conversation about this with her.

I created this issue so as to create a place to talk about the progress and any technical difficulties regarding upstream Haiku support.

**Comment by hallarempt** [comment]
I'll be watching the issue :-)

**Comment by n0toose** [comment]
Okay, as the licenses are incompatible (we use the MIT license), I don't think I can just carry over extrowerk's patches without permission.

Two plans:
- He gives us permission to create a derivative based around his work.
- I'd create a different port myself from the ground up, upstream.

**Comment by pulkomandy** [comment]
I don't think the MIT license we use here applies to patches to other software. This would be illegal in some cases (we package some non-free software for example). It seems kind of obvious that the patches are under the same license as the upstream they derive from. We should probably clarify that somewhere.

If you need explicit permission from @extrowerk I don't think he will mind. So why immediately consider the worse option of "let's redo the work needlessly"? That would be a stupid waste of time.

And even if that failed, the MIT license already grants a right to sublicense the software, so that wouldn't really be a problem, you could use that license clause to sublicense the patch into another license.

**Comment by hallarempt** [comment]
It's possible to integrate MIT licensed code in Krita. except if there's the advertisement clause.

**Comment by n0toose** [comment]
> Okay, as the licenses are incompatible (we use the MIT license)

Yeah, sorry, been very busy as of late IRL and misremembered something here/pulled that fact out of thin air. Thank you both for that quick fact-check.

**Comment by extrowerk** [comment]
> I don't think I can just carry over extrowerk's patches without permission.

Feel free to use my patches as you wish.

**Comment by bitigchi** [comment]
Did anyone succeed in re-building Krita 5.1.5? I'm afraid there'd be a tonne of dependencies to update.

**Comment by robxnano** [comment]
Yes, but the resulting application crashed on launch. The debug report wasn't particularly helpful (only showing the segfault occurred in `KisSpinboxColorSelector::updateSpinboxesWithNewValues()`) and I couldn't get any more information out of the debugger. Version 5.0.8 was the same. If anyone with Haiku debugging experience wants to try fixing it I'll make a branch with the updated patches.

There was only one problem with the dependencies. Although CMakeLists only requires Boost 1.55, it uses the Variant2 library so it actually needs 1.71. I had to update the Boost recipe to 1.81 to be able to compile it.

**Comment by pulkomandy** [comment]
For more debug info you may need to build things with the -gdwarf3 debug flag, I think support for newer formats is not yet complete in Debugger.

**Comment by hallarempt** [comment]
Ah, and for 5.2 the deps situation is going to be more interesting, too.

**Comment by Begasus** [comment]
@hallarempt I think this can be closed? I guess most is upstreamed that is really required, at least on our side only one patch is needed but is not related to Krita. ;)

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ac1f8c..2a63ba7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,7 +82,7 @@ if (POLICY CMP0167)
     cmake_policy(SET CMP0167 OLD)
 endif()
 
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+set(CMAKE_POSITION_INDEPENDENT_CODE OFF)
 
 if (APPLE)
     add_subdirectory(packaging/macos)
```
